### PR TITLE
NAS-125878 / 24.04 / Never run more than 8 disk temperature queries in parallel

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 0:15

**Comment by yocalebo** [comment]
I have a POC for querying ATA drives and their temperatures using ctypes and ioctl. Need to add the equivalent for SCSI as well. Once I'm confident in the code, we should switch to this instead of doing subprocess.

**Comment by themylogin** [comment]
@yocalebo so closing this and all related bugs?

**Comment by yocalebo** [comment]
> @yocalebo so closing this and all related bugs?

no, sorry. I'd still like to use this as a fallback in case of error and I'm not comfortable pushing the code in its current state.

File: src/middlewared/middlewared/utils/asyncio_.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import asyncio
 
 
-async def asyncio_map(func, arguments, limit=None):
-    semaphore = None
-    if limit is not None:
-        semaphore = asyncio.BoundedSemaphore(limit)
+async def asyncio_map(func, arguments, limit=None, *, semaphore=None):
+    if limit is not None or semaphore is not None:
```
Comment (sonicaj) [inline src/middlewared/middlewared/utils/asyncio_.py:5]: Should we raise an exception if both limit and semaphore are specified as `limit` would be redundant in that case ?

**Review by sonicaj — APPROVED** [review]

**Pull request merged**